### ISSUE TITLE
Reporting

### ISSUE THREAD

**Issue by chrisjvargo** [scott2b/surveyor#6]
Please don't worry about this until AFTER your vacation. Please. We don't want to rush this, and we are so close. I just did the math, to get these all annotated, it's going to cost a ton anyway, and I need to wait until the Fall to get the early fall to get the extra money.

So, right now this survey has 8 users that we're set to generate a report for:
http://surveyor.codeslick.com/survey/9/stats

So, each one saw 48 tweets so the minimum score for each user would be 48 (if they saw 48 tweets, and didn't annotate any of them). Yet, only A1XAOEDK0Y2KBT scored above 48. So something in the reporting must be off.

**Comment by chrisjvargo** [comment]
"minimum score" = sum of a user column in the reporting.

**Comment by scott2b** [comment]
I don't see any scores that come to 48. The scores are normalized to 0.0 - 1.0. Is it still the case that you want minimum score to be the sum o fa user column? I'm not sure I quite understand what that is.

**Comment by chrisjvargo** [comment]
We're talking about two different things. 

Download the CSV report for survey 9:
http://surveyor.codeslick.com/survey-report/9

If you open it in a spreadsheet editor, Column D (user ADVIE0ZHLWDDH) has 1s and 2s for each question. If we summed those up, the minimum score for each user would be 48 (if they saw 48 tweets, and didn't annotate any of them). Yet, the score is 45 for ADVIE0ZHLWDDH, 32 for Column E, and so on...

**Comment by scott2b** [comment]
Not everything gets marked with 1s and 2s. There is also a 0 condition, which I think means that the user did not see that tweet. I don't remember - but I also don't know if that has any bearing on this.

**Comment by chrisjvargo** [comment]
Right, but a user sees a minimum of 48 tweets, so the min score should be 48.

**Comment by scott2b** [comment]
what if a quota is filled? I'm not really sure that is the case here, but wouldn't that result in fewer tweets getting to the user?

**Comment by chrisjvargo** [comment]
It would. But that isn't the case here (Quota = 3) Tweets = 4,500. Total number of surveys to serve would be 13,000/45.

**Comment by chrisjvargo** [comment]
By chance, after three surveys we could have started to fill quotas. But no score was above 45. At least the first three should have been?

**Comment by scott2b** [comment]
That logic seems sound to me. I'm sure there is still some issue with the way I am putting the report together. The data model design as I have it is not ideal for the way we are doing this. I don't want to change the data model, but I will dig into the reporting code and see what I can find.

**Comment by scott2b** [comment]
remind me again what 0,1, and 2 indicate

**Comment by chrisjvargo** [comment]
0 = User did not see tweet
1 = User saw tweet, but did not "check"/annotated it
2 = User saw it, "checked"/annotated it

Chris J. Vargo
University of Alabama
Assistant Professor, Ph.D.

Advertising & Public Relations
------------------------------

[image: Twitter] <http://www.twitter.com/chrisjvargo> [image: LinkedIn]
<http://www.linkedin.com/in/chrisjvargo> [image: Google Plus]
<https://plus.google.com/105471295838192826025> [image: WordPress]
<http://www.chrisjvargo.com/>

**Comment by scott2b** [comment]
Ok, it seems like the problem is not really the code, but that the server is timing out before the full csv is delivered. Because it is delivered as a stream, there is no error condition that results. The code may or may not be right. As is, it looks like the CSV is about 11M rather than about 200k, so it's a really big download. I might be able to extend the timeout on the server, but I'm not sure yet if that is the right solution. In the meantime, let's do a sanity check. I'm getting the following stats on the full CSV file:

133057 rows of data (7392 tweets * 9 annotations = 66,528, so there is a factor of 2x I haven't figured out yet)
sums for individual users are in the 500 range. Not sure if that is expected.

**Comment by scott2b** [comment]
Ah wait, if I open the full csv in Excel, I get a correct number of rows of 66,528. There is a newline in the tweets or something that threw off my wc -l.

Ok, so # of rows looks good (I think).
~500 per user also seems justified: 48 tweets x 9 annotations (1 point for each view) + 48 (a second point for each tweet assuming a user gives a single annotation to each tweet) == 480

So, let me think about how to handle this long download.

**Comment by chrisjvargo** [comment]
Shot in the dark, maybe zipping it?

**Comment by scott2b** [comment]
yeah, zipping will help in terms of the actual download itself, but it is not an immediately implementable solution. Problem is that the csv comes down as a stream. So you are actually downloading the rows of the csv as they are generated. There is a fair amount of heavy querying going on. This is why I did it this way. I actually thought this would prevent a timeout issue, because the stream should stay open until explicitly closed. 

Even if I can fix this in the server configs this to keep the stream open, I'm not sure downloading an uncompressed 11M csv is the right thing. I think I will need to look into an async/offline approach. You click a button to generate the report, and somehow get a message when it is ready, with a link to the report. It's a fair amount of more work than just opening a stream and sending it back, but I think it is getting to where this is really the right way to do it.

**Comment by chrisjvargo** [comment]
I think we're good! Counts for the first user (ADVIE0ZHLWDDH):

0 66096
1 348
2 84

The counts add up to 432 (total annotations possible) / 9 (annotations per
question) = 48 (questions).
Works for A2S3MXU8T2I5H0 as well!

0 66096
1 354
2 78

Chris J. Vargo
University of Alabama
Assistant Professor, Ph.D.

Advertising & Public Relations
------------------------------

[image: Twitter] <http://www.twitter.com/chrisjvargo> [image: LinkedIn]
<http://www.linkedin.com/in/chrisjvargo> [image: Google Plus]
<https://plus.google.com/105471295838192826025> [image: WordPress]
<http://www.chrisjvargo.com/>
